src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs

```diff
@@ -218,12 +218,17 @@ public override string ToString()
         }
 
         // internal test hook
-        private static bool s_forceNullEntryPoint;
+        private static Assembly? s_overriddenEntryAssembly;
+
+        public static void SetEntryAssembly(Assembly newAssembly)
+        {
+            s_overriddenEntryAssembly = newAssembly;
```

Comment:
Do we need any validation. We discussed at least validating it's a runtime-defined assembly: https://github.com/dotnet/runtime/issues/101616#issuecomment-2096031589